src/utils/common-expression-helpers.ts

```diff
@@ -79,9 +79,13 @@ export class CommonExpressionHelpers {
   };
 
   addDaysToDate = (date, days) => {
-    date.setDate(date.getDate() + days);
-
-    return date;
+    const newDate = new Date(date);
+    dayjs(newDate).format('DD/MM/YYYY')
+    newDate.setDate(newDate.getDate() + days); 
```

Comment:
Since we are already using dayjs library could we use this

```
const newDate = dayjs(date).add(days, 'day');
return newDate.format('DD/MM/YYYY');
```

Reply:
The issue with the function was that it modifies the original date object directly using the setDate method. This lead to unexpected behavior because the modification affects the original object.

To fix this issue, you should create a new Date object based on the original date and perform the calculations on the new object.

To ensure proper calculations on the date I used the dayjs library to format the date in the "DD/MM/YYYY" format, and adding the specified number of days to the new date. Additionally, I included a check to verify if the resulting date is valid by checking if newDate.getTime() returns NaN. If the date is invalid, the function returns undefined and this was being displayed on the form, @samuelmale if this is not clear then we can have a session to explain better with an example